The job **`x86_64-msvc`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/9241517149/job/25423117088) [(plain)](https://github.com/rust-lang/rust/commit/fafa451987699e4f48ab4956b3783b342311ddb0/checks/25423117088/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at c70eae9a Merge fafa451987699e4f48ab4956b3783b342311ddb0 into bd184cc3e1ee16c1d74444fa246dcfbade9a31d5
[command]"C:\Program Files\Git\bin\git.exe" log -1 --format='%H'
'c70eae9a26b778ebbdada08cae32fbda87df205a'
'c70eae9a26b778ebbdada08cae32fbda87df205a'
##[group]Run echo "[CI_PR_NUMBER=$num]"
echo "[CI_PR_NUMBER=$num]"
---
installing mingw-w64-x86_64-libffi...
installing mingw-w64-x86_64-mpdecimal...
installing mingw-w64-x86_64-openssl...
Optional dependencies for mingw-w64-x86_64-openssl
    mingw-w64-x86_64-ca-certificates [pending]
installing mingw-w64-x86_64-readline...
installing mingw-w64-x86_64-tcl...
installing mingw-w64-x86_64-sqlite3...
installing mingw-w64-x86_64-tk...
---
[170/3411] Building CXX object lib\TableGen\CMakeFiles\LLVMTableGen.dir\TGParser.cpp.obj
[171/3411] Building CXX object utils\TableGen\CMakeFiles\obj.LLVMTableGenCommon.dir\Attributes.cpp.obj
[172/3411] Building CXX object utils\TableGen\CMakeFiles\obj.LLVMTableGenCommon.dir\CodeGenIntrinsics.cpp.obj
[173/3411] Building RC object utils\TableGen\CMakeFiles\llvm-min-tblgen.dir\__\__\resources\windows_version_resource.rc.res
FAILED: utils/TableGen/CMakeFiles/llvm-min-tblgen.dir/__/__/resources/windows_version_resource.rc.res 
RC C:\a\rust\rust\src\llvm-project\llvm\resources\windows_version_resource.rc utils\TableGen\CMakeFiles\llvm-min-tblgen.dir\__\__\resources\windows_version_resource.rc.res.d utils\TableGen\CMakeFiles\llvm-min-tblgen.dir\__\__\resources\windows_version_resource.rc.res "" "C:/a/rust/rust/build/bootstrap/debug/sccache-plus-cl.exe" C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\rc.exe -DGTEST_HAS_RTTI=0 -DUNICODE -D_CRT_NONSTDC_NO_DEPRECATE -D_CRT_NONSTDC_NO_WARNINGS -D_CRT_SECURE_NO_DEPRECATE -D_CRT_SECURE_NO_WARNINGS -D_GLIBCXX_ASSERTIONS -D_HAS_EXCEPTIONS=0 -D_SCL_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS -D_UNICODE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -DRC_FILE_VERSION=\"0\" -DRC_INTERNAL_NAME=\"llvm-tblgen\" -DRC_PRODUCT_NAME=\"LLVM\" -DRC_PRODUCT_VERSION=\"0\" -DRC_VERSION_FIELD_1=18 -DRC_VERSION_FIELD_2=1 -DRC_VERSION_FIELD_3=6 -DRC_VERSION_FIELD_4=0 -I C:\a\rust\rust\build\x86_64-pc-windows-msvc\llvm\build\utils\TableGen -I C:\a\rust\rust\src\llvm-project\llvm\utils\TableGen -I C:\a\rust\rust\build\x86_64-pc-windows-msvc\llvm\build\include -I C:\a\rust\rust\src\llvm-project\llvm\include -DWIN32 /nologo /fo utils\TableGen\CMakeFiles\llvm-min-tblgen.dir\__\__\resources\windows_version_resource.rc.res C:\a\rust\rust\src\llvm-project\llvm\resources\windows_version_resource.rc
Microsoft (R) Windows (R) Resource Compiler Version 10.0.10011.16384
Copyright (C) Microsoft Corporation.  All rights reserved.





fatal error RC1107: invalid usage; use RC /? for Help

[174/3411] Linking CXX static library lib\LLVMTableGen.lib
[175/3411] Building CXX object utils\TableGen\CMakeFiles\obj.LLVMTableGenCommon.dir\DirectiveEmitter.cpp.obj
[176/3411] Building CXX object utils\TableGen\CMakeFiles\obj.LLVMTableGenCommon.dir\IntrinsicEmitter.cpp.obj
---
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
 finished in 50.589 seconds
##[endgroup]
Build completed unsuccessfully in 0:01:28
make: *** [Makefile:101: ci-msvc-ps1] Error 1
  network time: Sun, 26 May 2024 07:34:35 GMT
##[error]Process completed with exit code 2.
Post job cleanup.
[command]"C:\Program Files\Git\bin\git.exe" version
```

</details>

--- a/src/ci/scripts/install-mingw.sh
+++ b/src/ci/scripts/install-mingw.sh
@@ -12,7 +12,7 @@ source "$(cd "$(dirname "$0")" && pwd)/../shared.sh"
 MINGW_ARCHIVE_32="i686-12.2.0-release-posix-dwarf-rt_v10-rev0.7z"
 MINGW_ARCHIVE_64="x86_64-12.2.0-release-posix-seh-rt_v10-rev0.7z"
 
-if isWindows; then
+if isWindows && isKnownToBeMingwBuild; then
     case "${CI_JOB_NAME}" in
         *i686*)
             bits=32
@@ -39,10 +39,7 @@ if isWindows; then
     esac
 
     if [[ "${CUSTOM_MINGW:-0}" == 0 ]]; then
-        pacboy -S --noconfirm toolchain:p
-        # According to the comment in the Windows part of install-clang.sh, in the future we might
-        # want to do this instead:
-        # pacboy -S --noconfirm clang:p ...
+        pacman -S --noconfirm --needed mingw-w64-$arch-toolchain
     else
         mingw_dir="mingw${bits}"
 

--- a/src/ci/scripts/install-msys2.sh
+++ b/src/ci/scripts/install-msys2.sh
@@ -1,6 +1,6 @@
 #!/bin/bash
-# Clean up and prepare the MSYS2 installation. MSYS2 is needed primarily for
-# the test suite (run-make), but is also used by the MinGW toolchain for assembling things.
+# Clean up and prepare the MSYS2 installation.
+# MSYS2 is used by the MinGW toolchain for assembling things.
 
 set -euo pipefail
 IFS=$'\n\t'
@@ -24,38 +24,4 @@ if isWindows; then
     fi
     ciCommandAddPath "C:\\hostedtoolcache\\windows\\Python\\${native_python_version}\\x64"
     ciCommandAddPath "C:\\hostedtoolcache\\windows\\Python\\${native_python_version}\\x64\\Scripts"
-
-    # Install pacboy for easily installing packages
-    pacman -S --noconfirm pactoys
-
-    # Remove these pre-installed tools so we can't accidentally use them, because we are using the
-    # MSYS2 setup action versions instead. Because `rm -r`-ing them is slow, we mv them off path
-    # instead.
-    # Remove pre-installed version of MSYS2
-    echo "Cleaning up existing tools in PATH"
-    notpath="/c/NOT/ON/PATH/"
-    mkdir --parents "$notpath"
-    mv -t "$notpath" "/c/msys64/"
-    # Remove Strawberry Perl, which contains a version of mingw
-    mv -t "$notpath" "/c/Strawberry/"
-    # Remove these other copies of mingw, I don't even know where they come from.
-    mv -t "$notpath" "/c/mingw64/"
-    mv -t "$notpath" "/c/mingw32/"
-    echo "Finished cleaning up tools in PATH"
-
-    if isKnownToBeMingwBuild; then
-        # Use the mingw version of CMake for mingw builds.
-        # However, the MSVC build needs native CMake, as it fails with the mingw one.
-        # Delete native CMake
-        rm -r "/c/Program Files/CMake/"
-        # Install mingw-w64-$arch-cmake
-        pacboy -S --noconfirm cmake:p
-
-        # It would be nice to use MSYS's git in MinGW builds so that it's tested and known to
-        # work. But it makes everything extremely slow, so it's commented out for now.
-        # # Delete Windows-Git
-        # rm -r "/c/Program Files/Git/"
-        # # Install MSYS2 git
-        # pacman -S --noconfirm git
-    fi
 fi

